### DISCUSSION TITLE
add Range to Access-Control-Request-Headers

### DISCUSSION THREAD

**Discussion by dotnetCarpenter** [community/community#14214] [Other Feature Feedback, Questions, & Ideas]
While we wait for [Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1733981) and [WebKit](https://bugs.webkit.org/show_bug.cgi?id=231174) browsers to [add range to CORS-safelisted request-headers](https://github.com/whatwg/fetch/issues/1310) (already shipped in Chromium), could you please add `Range` to `Access-Control-Allow-Headers` on https://raw.githubusercontent.com?

So we do not have to download an entire file but can break it up in ranges. I have [tested](https://dotnetcarpenter.github.io/bug-firefox-fetch-cors/) it in Chrome 99 and via nodejs, and your Varnish server supports fetching ranges.

Thank you in advance!

**Comment by dotnetCarpenter** [comment]
If you could please also add `accept-ranges` to `Access-Control-Expose-Headers` for `HEAD` requests? So that we can detect if a resource does indeed support `Range` before we start streaming data from raw.githubusercontent.com.

It is not strictly necessary for streaming  data from raw.githubusercontent.com in a browser but it prevent us from using libraries that would try to auto-detect if streaming is supported.
**Without** `Access-Control-Expose-Headers: accept-ranges` one would have to know that github supports streaming on raw.githubusercontent.com and hardcode it.

Currently is seems there is no `Access-Control-Expose-Headers` header for HEAD requests.
Preferred header for HEAD requests: `Access-Control-Expose-Headers: accept-ranges`

The following headers are always allowed and can be ignored: Cache-Control, Content-Language, Content-Length, Content-Type, Expires, Last-Modified and Pragma.

**Reply by ketourneau** [reply]
Any news on this request ? We need to do some range request to load point cloud from https://raw.githubusercontent.com.
